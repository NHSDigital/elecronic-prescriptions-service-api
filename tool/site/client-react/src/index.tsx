import {Bundle} from "fhir/r4"
import * as React from "react"
import {
  createPrescriptionSummaryProps,
  PrescriptionSummary
} from "./components/prescription-summary/prescriptionSummary"
import {PageContainer} from "./components/pageContainer"
import ReactDOM = require("react-dom")

(async function() {
  const baseUrl = "/"
  const urlParams = new URLSearchParams(window.location.search)
  const prescriptionId = urlParams.get("prescription_id")
  const getPrescriptionResponse: Response = await fetch(`${baseUrl}prescription/${prescriptionId}`)
  const bundle: Bundle = await getPrescriptionResponse.json()
  const prescriptionSummaryProps = createPrescriptionSummaryProps(bundle)
  const content = (
    <PageContainer>
      <PrescriptionSummary
        patient={prescriptionSummaryProps.patient}
        practitionerRole={prescriptionSummaryProps.practitionerRole}
      />
    </PageContainer>
  )
  ReactDOM.render(content, document.getElementById("root"))
}())
