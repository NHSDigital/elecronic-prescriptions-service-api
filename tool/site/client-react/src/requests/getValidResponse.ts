import {AxiosResponse} from "axios"
import {OperationOutcome} from "fhir/r4"
import {UnhandledAxiosResponseError} from "./unhandledAxiosResponseError"

export function getResponseDataIfValid<T>(
  response: AxiosResponse<T | OperationOutcome>,
  typeGuard: (unknown: unknown) => unknown is T
): T {
  const responseData = response.data
  if (responseData && typeGuard(responseData)) {
    return responseData
  }

  throw new UnhandledAxiosResponseError("Unexpected response from server.", response)
}
