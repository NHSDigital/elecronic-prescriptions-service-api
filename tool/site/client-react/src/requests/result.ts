import {FhirResource} from "fhir/r4"

export function isResult(data: unknown): data is Result {
  const result = data as Result
  return typeof result.success === "boolean"
    && "request" in result
    && "request_xml" in result
    && "response" in result
    && "response_xml" in result
}

export interface Result {
  success: boolean
  request: FhirResource
  request_xml: string
  response: FhirResource
  response_xml: string
}
