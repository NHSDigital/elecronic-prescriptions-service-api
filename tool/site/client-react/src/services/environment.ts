export type Environment = "internal-dev" | "int"

export const internalDev = "internal-dev"
export const int = "int"

export function isDev(environment: Environment): boolean {
  return environment === internalDev
}

export function isInt(environment: Environment): boolean {
  return environment === int
}
