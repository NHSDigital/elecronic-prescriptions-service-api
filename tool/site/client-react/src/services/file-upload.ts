import {Dispatch, SetStateAction} from "react"

export function readPrescriptionsFromFiles(
  files: FileList,
  prescriptionFilesUploaded: any[],
  setPrescriptionFilesUploaded: Dispatch<SetStateAction<any[]>>
): void {
  for (let i = 0; i < files.length; i++) {
    const reader = new FileReader()
    reader.onload = event => {
      const result = event.target.result
      prescriptionFilesUploaded.push(result)
      setPrescriptionFilesUploaded(prescriptionFilesUploaded)
    }
    reader.readAsText(files[i])
  }
}
