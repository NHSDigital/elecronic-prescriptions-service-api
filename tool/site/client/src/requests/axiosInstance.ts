import axios from "axios"

export const axiosInstance = axios.create({
  validateStatus: () => true
})

// todo: refresh token if required
axiosInstance.interceptors.request.use(function (config) {
  return config
}, function (error) {
  return Promise.reject(error)
})
