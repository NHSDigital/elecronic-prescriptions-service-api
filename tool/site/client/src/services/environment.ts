export type Environment = "internal-dev" | "internal-dev-sandbox" | "internal-qa" | "int" | "sandbox"

export const internalDev = "internal-dev"
export const internalDevSandbox = "internal-dev-sandbox"
export const internalQa = "internal-qa"
export const int = "int"
export const sandbox = "sandbox"

export function isDev(environment: Environment): boolean {
  return environment === internalDev || environment === internalDevSandbox
}

export function isQa(environment: Environment): boolean {
  return environment === internalQa
}

export function isInt(environment: Environment): boolean {
  return environment === int
}

export function isSandbox(environment: Environment): boolean {
  return environment === sandbox
}
