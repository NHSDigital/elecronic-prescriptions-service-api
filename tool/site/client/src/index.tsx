import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
