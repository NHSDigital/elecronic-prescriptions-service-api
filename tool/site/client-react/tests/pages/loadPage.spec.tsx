import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import userEvent from "@testing-library/user-event"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDev} from "../../src/services/environment"
import LoadPage from "../../src/pages/loadPage"

const baseUrl = "baseUrl/"
const context: AppContextValue = {baseUrl, environment: internalDev}

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays load page", async () => {
  const container = await renderPage()

  expect(screen.getByText("Load prescription(s)")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays custom fields when custom button is pressed", async () => {
  const container = await renderPage()
  userEvent.click(screen.getByText("Custom"))
  await waitFor(() => screen.getByText("Paste a FHIR prescription"))
  expect(screen.getByText("Paste a FHIR prescription")).toBeTruthy()
  expect(screen.getByText("Upload Test Pack")).toBeTruthy()
  expect(screen.getByText("Upload FHIR prescription files")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<LoadPage/>, context)
  await waitFor(() => screen.getByText("Load prescription(s)"))
  return container
}
