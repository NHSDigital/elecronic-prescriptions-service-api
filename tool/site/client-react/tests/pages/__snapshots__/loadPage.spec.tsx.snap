// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays custom fields when custom button is pressed 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Load prescription(s)</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label nhsuk-label--s\\" id=\\"prescriptionPath--label\\" for=\\"prescriptionPath\\">Select a prescription to load</label>
        <div class=\\"nhsuk-radios\\" id=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath--label\\" value=\\"custom\\" role=\\"radiogroup\\">
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-1\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-1--label\\" type=\\"radio\\" value=\\"primary-care/acute/nominated-pharmacy/medical-prescriber\\" checked=\\"\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-1--label\\" for=\\"prescriptionPath-1\\">Primary Care - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-2\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-2--label\\" type=\\"radio\\" value=\\"primary-care/repeat-prescribing\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-2--label\\" for=\\"prescriptionPath-2\\">Primary Care - Repeat Prescribing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-3\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-3--label\\" type=\\"radio\\" value=\\"primary-care/repeat-dispensing/nominated-pharmacy/medical-prescriber/author/gmc/responsible-party/medication-list/din\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-3--label\\" for=\\"prescriptionPath-3\\">Primary Care - Repeat Dispensing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-4\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-4--label\\" type=\\"radio\\" value=\\"secondary-care/community/acute/nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-4--label\\" for=\\"prescriptionPath-4\\">Secondary Care - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-5\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-5--label\\" type=\\"radio\\" value=\\"secondary-care/community/acute/no-nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-5--label\\" for=\\"prescriptionPath-5\\">Secondary Care - Acute</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-6\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-6--label\\" type=\\"radio\\" value=\\"secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-6--label\\" for=\\"prescriptionPath-6\\">Secondary Care - Repeat Dispensing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-7\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-7--label\\" type=\\"radio\\" value=\\"secondary-care/homecare/acute/nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-7--label\\" for=\\"prescriptionPath-7\\">Homecare - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-8\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-8--label\\" type=\\"radio\\" value=\\"custom\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-8--label\\" for=\\"prescriptionPath-8\\">Custom</label></div>
        </div>
      </div><label class=\\"nhsuk-label\\">Paste a FHIR prescription</label>
      <div class=\\"nhsuk-form-group\\"><textarea class=\\"nhsuk-textarea\\" name=\\"prescriptionTextArea\\" id=\\"prescriptionTextArea\\" rows=\\"10\\"></textarea></div><label class=\\"nhsuk-label\\">Upload Test Pack</label>
      <div class=\\"nhsuk-form-group\\"><input class=\\"nhsuk-input\\" name=\\"input_3qzrs\\" id=\\"input_3qzrs\\" type=\\"file\\" accept=\\"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet\\"></div><label class=\\"nhsuk-label\\">Upload FHIR prescription files</label>
      <div class=\\"nhsuk-form-group\\"><input class=\\"nhsuk-input\\" name=\\"input_1kodh\\" id=\\"input_1kodh\\" type=\\"file\\" multiple=\\"\\" accept=\\"application/json\\"></div>
    </fieldset>
  </div>
  <div class=\\"sc-gsDKAQ fnrIHM\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">View</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>
</form>"
`;

exports[`Displays load page 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Load prescription(s)</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label nhsuk-label--s\\" id=\\"prescriptionPath--label\\" for=\\"prescriptionPath\\">Select a prescription to load</label>
        <div class=\\"nhsuk-radios\\" id=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath--label\\" value=\\"primary-care/acute/nominated-pharmacy/medical-prescriber\\" role=\\"radiogroup\\">
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-1\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-1--label\\" type=\\"radio\\" value=\\"primary-care/acute/nominated-pharmacy/medical-prescriber\\" checked=\\"\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-1--label\\" for=\\"prescriptionPath-1\\">Primary Care - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-2\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-2--label\\" type=\\"radio\\" value=\\"primary-care/repeat-prescribing\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-2--label\\" for=\\"prescriptionPath-2\\">Primary Care - Repeat Prescribing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-3\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-3--label\\" type=\\"radio\\" value=\\"primary-care/repeat-dispensing/nominated-pharmacy/medical-prescriber/author/gmc/responsible-party/medication-list/din\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-3--label\\" for=\\"prescriptionPath-3\\">Primary Care - Repeat Dispensing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-4\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-4--label\\" type=\\"radio\\" value=\\"secondary-care/community/acute/nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-4--label\\" for=\\"prescriptionPath-4\\">Secondary Care - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-5\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-5--label\\" type=\\"radio\\" value=\\"secondary-care/community/acute/no-nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-5--label\\" for=\\"prescriptionPath-5\\">Secondary Care - Acute</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-6\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-6--label\\" type=\\"radio\\" value=\\"secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-6--label\\" for=\\"prescriptionPath-6\\">Secondary Care - Repeat Dispensing (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-7\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-7--label\\" type=\\"radio\\" value=\\"secondary-care/homecare/acute/nominated-pharmacy/clinical-practitioner\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-7--label\\" for=\\"prescriptionPath-7\\">Homecare - Acute (nominated)</label></div>
          <div class=\\"nhsuk-radios__item\\"><input class=\\"nhsuk-radios__input\\" id=\\"prescriptionPath-8\\" name=\\"prescriptionPath\\" aria-labelledby=\\"prescriptionPath-8--label\\" type=\\"radio\\" value=\\"custom\\"><label class=\\"nhsuk-label nhsuk-radios__label\\" id=\\"prescriptionPath-8--label\\" for=\\"prescriptionPath-8\\">Custom</label></div>
        </div>
      </div>
    </fieldset>
  </div>
  <div class=\\"sc-gsDKAQ fnrIHM\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">View</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>
</form>"
`;
